### DISCUSSION TITLE
Dialog is sometimes not interactive on initial mount

### DISCUSSION THREAD

**Discussion by erm1116** [chakra-ui/ark#2852] [General]
### Description

In Dialog, elements may not become interactive when using Portal.
https://github.com/chakra-ui/ark/discussions/2295 seems to have the similar problem.

---

our dialog component implementation is as follows:
```tsx
    <ArkDialog.Root
      lazyMount
      unmountOnExit
      modal
      open={isOpen}
      onOpenChange={handleOpenChange}
    >
      <Portal>
        <ArkDialog.Backdrop className={classes.backdrop} />
        <ArkDialog.Positioner className={classes.positioner}>
          <ArkDialog.Content className={classes.container} ref={ref}>
            <ArkDialog.Title>
              {title}
            </ArkDialog.Title>
            <ArkDialog.CloseTrigger
              className={classes.closeTrigger}
            >
              <CloseIcon />
            </ArkDialog.CloseTrigger>
            {children}
          </ArkDialog.Content>
        </ArkDialog.Positioner>
      </Portal>
    </ArkDialog.Root>
```

If the Portal is not used, this problem will not occur, so there may be a race condition problem in the Portal.

### Link to Reproduction (or Detailed Explanation)

https://codesandbox.io/p/devbox/loving-morning-4vk2xj

### Steps to Reproduce

Unfortunately, the exact same problem was not reproduced on CodeSandbox.
But, if you click the reload button several times, you can encounter the problem of the Dialog closing without doing anything.
I think this is related to race conditions.

https://github.com/user-attachments/assets/b6fe4a8e-6d21-4f87-ac5f-e0e8cfe1ed59

---

The reproduction capture of using our Dialog implementation is below:

https://github.com/user-attachments/assets/35f45cc4-2eb0-4e79-84bd-370da3b832e3

### Ark UI Version

CodeSandbox: 3.10.0, Our implementation: 3.7.0

### Framework

- [X] React
- [ ] Solid
- [ ] Vue

### Browser

Google Chrome (Chromium 128.0)

### Additional Information

I've also posted it on discord.
https://discord.com/channels/964648323304808488/1273658223735476294

Perhaps, an Issue I created earlier is relevant.
https://github.com/chakra-ui/ark/issues/2744

**Comment by segunadebayo** [comment]
This is an interesting race condition within the Portal. I can't reproduce this on my end, though.

I recommend opening the dialog when the page mounts via a `useEffect` if it's a blocker.

> I can't think of a way to fix this since it's within React's timing.

**Reply by ivancuric** [reply]
Were you able to reproduce https://github.com/chakra-ui/ark/issues/2716#issuecomment-2330942693?

**Comment by segunadebayo** [comment]
Closing — use `defaultOpen` with `lazyMount` for dialogs that need to be open on mount.
